enhancements/update/versioning-cincinnati-api.md

```diff
@@ -0,0 +1,122 @@
+---
+title: versioning-cincinnati-api
+authors:
+- "@pratikmahajan"
+  reviewers:
+- @openshift/openshift-team-cincinnati-maintainers
+- @openshift/openshift-team-cincinnati
+- @openshift/team-cincinnati-operator
+  approvers:
+- @crawford
+- @LalatenduMohanty
+- @sdodson
+  creation-date: 2021-08-20
+  last-updated: 2021-08-20
+  status: implementable
+  see-also:
+-
+replaces: []
+superseded-by: []
+---
+
+# Versioning cincinnati api and json schema
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+As of today, we declare the schema version in Cincinnati Graph URI. We currently, do not embed the version information
+in json schema. The enhancement will embed the version information in the json schema. This has the benefit of being
+easy to implement and to pass around if we ship Cincinnati JSON over non-HTTP channels. The enhancement will also begin
+versioning Cincinnati payloads by media type, for example `application/vnd.redhat.cincinnati.v1+json`.
+The move to supplying requested version in media type field will require us to move towards a generic un-versioned Graph
+URI, with the move, we'll still support old URI for foreseeable future.
+
+## Motivation
+
+### Goals
+1. Embed version information in json schema.
+2. Implement media type versioning for clients requesting a specific schema version of Cincinnati
+3. Move to a generic un-versioned URI for Cincinnati
+4. Cincinnati continues to support older clients after adding support for new clients for foreseeable future.
+5. Add metrics to identify hits on URIs and versions.
+
+### Non-Goals
+
+## Proposal
+
+### Implementation
+
+#### Version information in json schema
+Embed the version information in json schema. The updated json schema should look like:
+```json
+{
+    "version": 1,
+    "nodes": [],
+    "edges": []
+}
+```
+
+#### Media Type Versioning
+Currently, the versioning is done by using the versioned URI and does not support media type versioning. The user should
+be able to request a specific version of cincinnati graph by supplying the version in media type field of the request.
+for example `application/vnd.redhat.cincinnati.v1+json`. Here are some of the
```

Comment:
I think you want to call out [the `Accept` header][1] explicitly in this section, to show that this is [proactive negotiation][2].  Something like:

```Markdown
## Media Type Versioning

Currently, the versioning is done by using the versioned URI and does not support media type versioning.
This enhancement allows users to request a specific version of the Cincinnati specification [proactively][proactive-negotiation] with [the `Accept` header][accept].
For example, `Accept: application/vnd.redhat.cincinnati.v1+json` says "I only understand v1 Cincinnati graphs; please give me that format if possible".

In the event where the media type is not provided, or `application/json` is provided, Cincinnati should return a default
response that is compatible with the oldest supported version so as not to break old clusters from upgrading.

In a similar space, the Open Container Initiative Image Format Specification [uses versioned media types][image-spec-media-types] to identify versions of configuration and other formats.

[accept]: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2
[image-spec-media-types]: https://github.com/opencontainers/image-spec/blob/v1.0.1/media-types.md
[proactive-negotiation]: https://datatracker.ietf.org/doc/html/rfc7231#section-3.4.1
```

[1]: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2
[2]: https://datatracker.ietf.org/doc/html/rfc7231#section-3.4.1